src/service/topics.ts

```diff
@@ -172,6 +172,19 @@ const inviteCommunityMember = async (topicId, memberIds) => {
   }
 };
 
+const putCommunityImage = async (topicName, icon, cover) => {
+  try {
+    const resApi = await api.put(`/topics/${topicName}`, {
+      icon,
+      cover
+    });
+    return resApi.data;
+  } catch (error) {
+    crashlytics().recordError(new Error(error));
+    return error.response.data;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `putCommunityImage` function does not handle errors properly. It returns the error response data directly, which might expose sensitive information. Instead, it should return a custom error message or an error object that doesn't contain sensitive data. Also, it's better to log the error for debugging purposes.

```diff
  try {
    const resApi = await api.put(`/topics/${topicName}`, {
      icon,
      cover
    });
    return resApi.data;
  } catch (error) {
    crashlytics().recordError(new Error(error));
-   return error.response.data;
+   console.error(error);
+   throw new Error('Failed to update community image');
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->